common/src/main/java/com/wynntils/models/items/annotators/game/MiscAnnotator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.items.annotators.game;
+
+import com.wynntils.handlers.item.ItemAnnotation;
+import com.wynntils.handlers.item.ItemAnnotator;
+import com.wynntils.models.items.items.game.MiscItem;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.mc.LoreUtils;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.world.item.ItemStack;
+
+public final class MiscAnnotator implements ItemAnnotator {
+    private static final String UNTRADABLE = "§cUntradable Item";
+    private static final String QUEST_ITEM = "§cQuest Item";
+
+    @Override
+    public ItemAnnotation getAnnotation(ItemStack itemStack, String name) {
+        ListTag loreTag = LoreUtils.getLoreTag(itemStack);
+        if (loreTag == null) return null;
+
+        boolean untradable = false;
+        boolean questItem = false;
+
+        for (Tag line : loreTag) {
+            String coded = ComponentUtils.getCoded(line.getAsString());
+            if (coded.equals(UNTRADABLE)) {
+                untradable = true;
+            }
+            if (coded.equals(QUEST_ITEM)) {
+                questItem = true;
+            }
```

Comment:
Afaik only either of these can happen, so we can break early.

Reply:
(`getCoded` is quite heavy from profiling, so this is not without basis)

Reply:
I am like 99% sure that you can have both. It is not common, but possible. I think I encountered some such items when doing the gear model.

Reply:
Okay.

Reply:
I also remember strongly that you were worried about item annotation performance, which infected me as well, so I measured very carefully. In the end, it was almost impossible to measure, except when like speed-clicking through the Trade Market pages. 

This is just before the fallback annotator. I don't think you have to be that worried about performance.